Sources/Utilities/AppQualifyingService.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import Networking
+import SecureStore
+
+protocol QualifyingService: AnyObject {
+    var delegate: AppQualifyingServiceDelegate? { get set }
+    func initiate()
+    func evaluateUser() async
+}
+
+enum AppInformationState {
+    case appOffline
+    case appInfoError
+    case appOutdated
+    case appUnconfirmed
+    case appConfirmed
+    // TODO: DCMAW-9824 - New state may be needed for appInfo.allowAppUsage which corresponds to 'available' in appInfo json
+}
+
+enum AppLocalAuthState: Equatable {
+    case userFailed(_ error: Error)
+    case userExpired
+    case userUnconfirmed
+    case userOneTime
+    case userConfirmed
+    
+    static func == (lhs: AppLocalAuthState, rhs: AppLocalAuthState) -> Bool {
+        switch (lhs, rhs) {
+        case (.userFailed, .userFailed),
+            (.userExpired, .userExpired),
+            (.userUnconfirmed, .userUnconfirmed),
+            (.userOneTime, .userOneTime),
+            (.userConfirmed, .userConfirmed):
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+final class AppQualifyingService: QualifyingService {
+    private let updateService: AppInformationServicing
+    private let sessionManager: SessionManager
+    weak var delegate: AppQualifyingServiceDelegate?
+    
+    private var appInfoState: AppInformationState = .appUnconfirmed {
+        didSet {
+            if appInfoState == .appOffline {
+                // Query cache?
+            }
+            delegate?.didChangeAppInfoState(state: appInfoState)
+        }
+    }
+    
+    private var userState: AppLocalAuthState = .userUnconfirmed {
+        didSet {
+            delegate?.didChangeUserState(state: userState)
+        }
+    }
```

Comment:
Using the delegate pattern these properties provide updates to their delegate class (the QualifyingCoordinator in this case) to guide the flow of app and user qualification. The app state process guides some navigation paths or, in a happy path scenario, falls through to the user state process which has it's own navigation paths.